app/src/main/java/com/kms/cura/model/UserModel.java

```diff
@@ -75,4 +78,20 @@ public void doctorSearch(DoctorSearchEntity entity) {
                 EntityToJsonConverter.convertEntityToJson(entity).toString(), doctorResponse);
         VolleyHelper.getInstance().addToRequestQueue(stringRequest, tag_string_req);
     }
+
+    public void updatePatientHealth(PatientUserEntity entity) throws Exception {
+        StringBuilder builder = new StringBuilder();
+        builder.append(Settings.SERVER_URL);
+        builder.append(Settings.UPDATE_PATIENT_HEALTH);
+        PatientModelResponse response =  new PatientModelResponse();
+        StringRequest stringRequest = RequestUtils.createRequest(builder.toString(), Request.Method.POST,
+                EntityToJsonConverter.convertEntityToJson(entity).toString(), response);
+        VolleyHelper.getInstance().addToRequestQueue(stringRequest, tag_string_req);
+        while (!response.isGotResponse());
+        if(!response.isResponseError()) {
+            CurrentUserProfile.getInstance().setData(response.getUpdatedPatient());
```

Comment:
Don't set the data here. Instead set it where UPDATE_PATIENT_HEALTH_SUCCESS is listened